browser/brave_origin/brave_origin_service_factory.cc

```diff
@@ -42,4 +47,99 @@ bool BraveOriginServiceFactory::ServiceIsCreatedWithBrowserContext() const {
+  const auto& policy_map = policy::kBraveSimplePolicyMap;
+
+  // Define BraveOrigin-specific metadata for each pref we want to control
+  static constexpr auto brave_origin_metadata =
+      base::MakeFixedFlatMap<std::string_view, BraveOriginPrefMetadata>({
+
+          // Wayback Machine preferences
+          {kBraveWaybackMachineEnabled,
+           BraveOriginPrefMetadata(false, BraveOriginPolicyScope::kProfile,
+                                   /*user_settable=*/true)},
+
+#if BUILDFLAG(ENABLE_TOR)
+          // Tor preferences
+          {tor::prefs::kTorDisabled,
+           BraveOriginPrefMetadata(true, BraveOriginPolicyScope::kGlobal,
+                                   /*user_settable=*/false)},
+#endif
+
+          // Stats reporting
+          {kStatsReportingEnabled,
+           BraveOriginPrefMetadata(false, BraveOriginPolicyScope::kGlobal,
+                                   /*user_settable=*/true)},
+
+          // P3A preferences
+          {p3a::kP3AEnabled,
+           BraveOriginPrefMetadata(false, BraveOriginPolicyScope::kGlobal,
+                                   /*user_settable=*/true)},
+
+          // Brave Rewards preferences
+          {brave_rewards::prefs::kDisabledByPolicy,
+           BraveOriginPrefMetadata(true, BraveOriginPolicyScope::kProfile,
+                                   /*user_settable=*/false)},
+
+          // Brave Wallet preferences
+          {brave_wallet::prefs::kDisabledByPolicy,
+           BraveOriginPrefMetadata(true, BraveOriginPolicyScope::kProfile,
+                                   /*user_settable=*/false)},
+
+          // AI Chat preferences
+          {ai_chat::prefs::kEnabledByPolicy,
+           BraveOriginPrefMetadata(false, BraveOriginPolicyScope::kProfile,
+                                   /*user_settable=*/false)},
+
+#if BUILDFLAG(ENABLE_SPEEDREADER)
+          // Speedreader preferences
+          {speedreader::kSpeedreaderPrefFeatureEnabled,
+           BraveOriginPrefMetadata(false, BraveOriginPolicyScope::kProfile,
+                                   /*user_settable=*/true)},
+#endif
+
+          // Brave News preferences
+          {brave_news::prefs::kBraveNewsDisabledByPolicy,
+           BraveOriginPrefMetadata(true, BraveOriginPolicyScope::kProfile,
+                                   /*user_settable=*/false)},
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+          // Brave VPN preferences
+          {brave_vpn::prefs::kManagedBraveVPNDisabled,
+           BraveOriginPrefMetadata(true, BraveOriginPolicyScope::kProfile,
+                                   /*user_settable=*/false)},
+#endif
+
+          // Brave Talk preferences
+          {kBraveTalkDisabledByPolicy,
+           BraveOriginPrefMetadata(true, BraveOriginPolicyScope::kProfile,
+                                   /*user_settable=*/false)},
+      });
+
+  // Build the final preference definitions
+  for (const auto& [policy_key, pref_name, type] : policy_map) {
```

Comment:
nit: should we iterate `policy::kBraveSimplePolicyMap` directly, and get rid of the `policy_map` variable?

Reply:
Fixed in rebase